added battery soc sensors

Man-in-Black opened the pull request:
I wanted to know the SoC of my battery and added the corresponding sensors which I read from the modbus. I think its only one of many sensors reporting the SoC, but it works fine.

Man-in-Black commented:
Here also a screenshot of the new sensor:
<img width="287" height="250" alt="image" src="https://github.com/user-attachments/assets/09ad5dc0-d5dd-4371-ac32-5f5e23c233e0" />

unbekannt3 commented:
Did the same on my HA instance here but also added a variable icon to the SoC sensor in sensor.py around line 176 beneath _handle_coordinator_update:
```
    @property
    def icon(self) -> str | None:
        """Return the icon to use in the frontend."""
        # battery soc sensor
        if self._sensor_key == "battery_soc":
            try:
                soc = int(self.native_value)
            except (ValueError, TypeError):
                return "mdi:battery-unknown"

            if soc > 95:
                return "mdi:battery"
            
            rounded_soc = (soc + 5) // 10 * 10
            if rounded_soc == 0:
                return "mdi:battery-outline"
            
            return f"mdi:battery-{rounded_soc}"
        
        # For all other sensors return the static icon defined in const.py
        return self._definition.get("icon")
```

Man-in-Black commented:
ok so you're changing the icon, kinda cool.

Shall I add this to this pull request?

Man-in-Black commented:
@unbekannt3 I added your code to the pull request and I also changed the requirements in the manifest.py, because with == it didn't start on my HA after I updated it to 2025.10.0
